enumerable_methods.rb

```diff
@@ -0,0 +1,132 @@
+module Enumerable
+  # start of my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+  # end of my_each
+
+  # start of my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+  # end of my_each_with_index
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+
+    until i >= length
+      new_array << i if yield self[i]
+      i += 1
+    end
+  end
+  # end of my_select
+
+  # my_all?
+  def my_all?
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+  # end of my_all?
+
+  # my_any?
+  def my_any?
+    my_each do |item|
+      return true if yield item
+    end
+    false
+  end
+  # end of my_any?
+
+  # my_none?
+  def my_none?
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
+  # end of my_none?
+
+  # my_count
+  def my_count
+    count = 0
+    my_each do |_item|
+      count += 1
+    end
+    count
+  end
+  # end of my_count
+
+  # start of my_map
+  def my_map(_proc = nil)
```

Comment:
- [ ] [`map`](https://rubyapi.org/2.7/o/enumerable#method-i-map) Returns a new array with the results of running block once for every element in enum.

Your current implementation is returning the same array regarding the passed block.

```ruby
[5, 2, 1].my_map { |x| x * 2 }
# should give [10, 2, 1]
# But the result is [5, 2, 1]
```

You're appending it to the array in `new_array << item` and ignoring the passed block. Let's append yielded item from the block.